Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/Doctrine/DecendantOfTypeConditionGenerator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Neos\ContentRepository\Security\Authorization\Privilege\Node\Doctrine;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\Persistence\Mapping\ClassMetadata;
+use Doctrine\ORM\Query\Filter\SQLFilter as DoctrineSqlFilter;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * A SQL generator to create a condition matching anything.
+ */
+class DecendantOfTypeConditionGenerator implements SqlGeneratorInterface
+{
+    private array $nodetypes;
+
+    /**
+     * @param array $nodetypes
+     */
+    public function __construct(array $nodetypes)
+    {
+        $this->nodetypes = $nodetypes;
+    }
+
+    /**
+     * Returns an SQL query part that is basically a no-op in order to match any entity
+     *
+     * @param DoctrineSqlFilter $sqlFilter
+     * @param ClassMetadata $targetEntity
+     * @param string $targetTableAlias
+     * @return string
+     */
+    public function getSql(DoctrineSqlFilter $sqlFilter, ClassMetadata $targetEntity, $targetTableAlias)
+    {
+        $nodetypeList = implode("','", $this->nodetypes);
+
+        return "select * from public.neos_contentrepository_domain_model_nodedata n1
+        JOIN public.neos_contentrepository_domain_model_nodedata n2 ON n1.path LIKE CONCAT('%', n2.path, '%')
```

Comment:
hmm, CONCAT('%, n2.path, '%') adds a % before the n2.path; IMHO this is wrong; because we want prefix matching. Or am I wrong? :)

Reply:
Yea you are right, I was never really that good at writing SQL queries, but I Tested both queries, and they return the same result.